### PR TITLE
Add Perfetto

### PR DESCRIPTION
**Pull request by kaidokert**
Upstream version from Chrome m114, tag 114.0.5735.356

b/319086452

### CODE REVIEW

**Review by andrewsavage1 — APPROVED** [review]

File: third_party/perfetto/METADATA
Comment (dahlstrom-g) [inline third_party/perfetto/METADATA:1]: An issue I've mentioned with reusing `METADATA` files for our own peculiar purposes—
there's one here already:
https://android.googlesource.com/platform/external/perfetto/+/f2da6df2/METADATA

File: third_party/perfetto/METADATA [hunk 1]
```diff
@@ -1,3 +1,22 @@
+name: "perfetto"
+description:
+  "Perfetto system profiling and tracing"
+
 third_party {
+  identifier {
+    type: "ChromiumVersion"
+    value: "114.0.5735.338"  # from https://chromereleases.googleblog.com/2024/03/long-term-support-channel-update-for.html
```
Comment (dahlstrom-g) [inline third_party/perfetto/METADATA:8]: The blog URL shows 114.0.5735.355, not 114.0.5735.338.

File: third_party/perfetto/METADATA [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+  }
+  identifier {
+    type: "Git"
+    value: "https://android.googlesource.com/platform/external/perfetto.git"
+    version: "f2da6df2f144e41e1c1428f11e8b388eaf8a2209"
+    # from https://chromium.googlesource.com/chromium/src/+/refs/tags/114.0.5735.338/DEPS#1698
```
Comment (dahlstrom-g) [inline third_party/perfetto/METADATA:14]: perhaps
https://chromium.googlesource.com/chromium/src/+/114.0.5735.355/DEPS#1698

File: third_party/perfetto/METADATA (hunk 1 above)
Comment (kaidokert) [reply third_party/perfetto/METADATA:8]: Yes but the tag exists for the newer commit too. I grabbed the latest tagged one from m114 release branch. Is there a reason why we'd want to pin to slightly older one ?

File: third_party/perfetto/METADATA (hunk 2 above)
Comment (kaidokert) [reply third_party/perfetto/METADATA]: Updated to shorter link, but left hte 388 in there - unless there's a strong reason to pick older ?

File: third_party/perfetto/METADATA
Comment (kaidokert) [reply third_party/perfetto/METADATA:1]: Yes, this is the first time we ran into a real conflict. It's okay in this case as we don't overload or change anything in what upstream provided. But in long term i'm expecting more problems.

File: third_party/perfetto/METADATA (hunk 1 above)
Comment (dahlstrom-g) [reply third_party/perfetto/METADATA:8]: > Yes but the tag exists for the newer commit too.

What tag? What newer commit?

> I grabbed the latest tagged one from m114 release branch. Is there a reason why we'd want to pin to slightly older one ?

To expand on what I wrote above, the blog post states:
“LTS-114 is being updated in the LTS channel to 114.0.5735.355”.
https://chromereleases.googleblog.com/2024/03/long-term-support-channel-update-for.html

File: third_party/perfetto/METADATA (hunk 2 above)
Comment (dahlstrom-g) [reply third_party/perfetto/METADATA]: Perhaps this explains the confusion: 338 looks like 388.

File: third_party/perfetto/METADATA (hunk 1 above)
Comment (kaidokert) [reply third_party/perfetto/METADATA]: Nevermind - removing the URL as it has no significance. Updated the tag to `114.0.5735.356` which is latest available on `m114`

File: third_party/perfetto/METADATA (hunk 2 above)
Comment (kaidokert) [reply third_party/perfetto/METADATA]: Updated to `114.0.5735.356`, i.e. latest tag available - the imported hash doesn't change.

File: third_party/perfetto/METADATA (hunk 1 above)
Comment (dahlstrom-g) [reply third_party/perfetto/METADATA]: The blog post is is significant in that it identifies a release
as mentioned more than one place in our documentation.
The tag `114.0.5735.356` does not correspond to a release.

**Pull request merged**